$OpenBSD: patch-ssl_sslprint_c,v 1.1 2013/04/25 15:21:40 sthen Exp $

http://sourceforge.net/p/ssldump/patches/7/

ssldump segfaults while parsing a certificate with unknown type.
This is because the loop in ssl_print_enum (ssl/sslprint.c) checks
against the wrong terminating value for client_certificate_type_decoder
(ssl/ssl.enums.c).

--- ssl/sslprint.c.orig	Thu Apr 25 16:09:58 2013
+++ ssl/sslprint.c	Thu Apr 25 16:10:07 2013
@@ -406,7 +406,7 @@ int ssl_print_enum(ssl,name,dtable,value)
     if(name) explain(ssl,"%s ",name);    
     INDENT;
     
-    while(dtable && dtable->type!=-1){
+    while(dtable && dtable->type!=0){
       if(dtable->type == value){
         INDENT_INCR;
         explain(ssl,"%s",dtable->name);
